code/modules/clothing/spacesuits/flightsuit.dm

```diff
@@ -0,0 +1,1099 @@
+		if(crash_disabled && (crash_damage <= 1))
+			crash_disabled = 0
+			crash_disable_message = 0
+			usermessage("Flightpack: Stabilizers re-calibrated. Flightpack re-enabled.")
+	if(emp_damage)
+		emp_damage = Clamp(emp_damage-emp_heal_amount, 0, emp_disable_threshold * 10)
+		if(emp_damage >= emp_disable_threshold)
+			emp_disabled = 1
+		if(emp_disabled && (emp_damage <= 0.5))
+			emp_disabled = 0
+			usermessage("Flightpack: Systems rebooted. Flightpack re-enabled")
+	disabled = crash_disabled + emp_disabled
+	if(disabled)
+		if(crash_disabled && (!crash_disable_message))
+			usermessage("WARNING: STABILIZERS DAMAGED. UNABLE TO CONTINUE OPERATION. PLEASE WAIT FOR AUTOMATIC RECALIBRATION", 1)
+			usermessage("Your flightpack abruptly shuts off!", 2)
+			crash_disable_message = 1
+		if(emp_disabled)
+			usermessage("WARNING: POWER SURGE DETECTED FROM INTERNAL SHORT CIRCUIT. PLEASE WAIT FOR AUTOMATIC REBOOT.", 1)
+			usermessage("Your flightpack abruptly shuts off!", 2)
+			emp_disable_message = 1
+		if(flight)
+			disable_flight(1)
+
+/obj/item/device/flightpack/update_icon()
+	if(!flight)
+		icon_state = initial(icon_state)
+		item_state = initial(item_state)
+	if(flight)
+		icon_state = icon_state_active
+		item_state = item_state_active
+		if(boost)
+			icon_state = icon_state_boost
+			item_state = item_state_boost
+	if(wearer)
+		wearer.update_inv_wear_suit()
+	..()
+
+/obj/item/device/flightpack/proc/handle_flight()
+	if(!flight)
+		return 0
+	wearer.float(2)
+
+/obj/item/device/flightpack/proc/handle_boost()
+	if(boost)
+		boost_charge = Clamp(boost_charge-boost_drain, 0, boost_maxcharge)
+		if(boost_charge < 1)
+			deactivate_booster()
+	if(boost_charge < boost_maxcharge)
+		boost_charge = Clamp(boost_charge+boost_chargerate, 0, boost_maxcharge)
+
+
+/obj/item/device/flightpack/proc/cycle_power()
+	if(powersetting < powersetting_high)
+		powersetting++
+	else
+		powersetting = 1
+	momentum_gain = powersetting * 10
+	if(suit)
+		if(suit.user)
+			wearer << "<span class='notice'>FLIGHTPACK: Engines set to force [momentum_gain].</span>"
+
+/obj/item/device/flightpack/proc/flight_impact(atom/movable/victim)	//Yes, victim.
+	var/density = 0
+	var/anchored = 1
+	if(isclosedturf(victim))
+		density = 1
+		anchored = 1
+	else if(istype(victim,))
+		anchored = victim.anchored
+	if(!anchored)
+		var/turf/target = get_edge_target_turf(victim, suit.user.dir)
+		target = get_edge_target_turf(target, suit.user.dir)	//Woop
+		var/aim = rand(1, 5)
+		switch(aim)
+			if(1)
+				target = get_step(target, NORTH)
+				target = get_step(target, NORTH)
+				target = get_step(target, EAST)
+				target = get_step(target, EAST)
+			if(2)
+				target = get_step(target, NORTH)
+				target = get_step(target, NORTH)
+				target = get_step(target, WEST)
+				target = get_step(target, WEST)
+			if(3)
+				target = get_step(target, SOUTH)
+				target = get_step(target, SOUTH)
+				target = get_step(target, EAST)
+				target = get_step(target, EAST)
```

Comment:
Each of these doubled get_steps could be simplified into the following;

    for(var/i in 1 to 2)
    	target = get_step(target, WHATEVER DIR HERE)